match/new_orleans_da.py

```diff
@@ -0,0 +1,129 @@
+from datamatch import ThresholdMatcher, JaroWinklerSimilarity, ColumnsIndex
+from lib.path import data_file_path, ensure_data_dir
+from lib.clean import float_to_int_str
+import pandas as pd
+import sys
+sys.path.append('../')
+
+
+def prepare_new_orleans_so_post_data():
+    post = pd.read_csv(data_file_path('clean/pprr_post_2020_11_06.csv'))
+    return post[post.agency == 'orleans parish so']
+
+
+def prepare_new_orleans_pd_post_data():
+    post = pd.read_csv(data_file_path('clean/pprr_post_2020_11_06.csv'))
+    return post[post.agency == 'new orleans pd']
+
+
+def match_cprr_and_noso_post(cprr, post):
+    dfa = cprr[['uid', 'first_name', 'last_name']]
+    dfa.loc[:, 'fc'] = dfa.first_name.map(lambda x: x[:1])
+    dfa = dfa.drop_duplicates(subset=['uid']).set_index('uid')
+
+    dfb = post[['uid', 'first_name', 'last_name']]
+    dfb.loc[:, 'fc'] = dfb.first_name.map(lambda x: x[:1])
+    dfb = dfb.drop_duplicates(subset=['uid']).set_index('uid')
+
+    matcher = ThresholdMatcher(ColumnsIndex(['fc']), {
+        'last_name': JaroWinklerSimilarity(),
+        'first_name': JaroWinklerSimilarity(),
+    }, dfa, dfb)
+    decision = 1
+    matcher.save_pairs_to_excel(data_file_path(
+        'match/new_orleans_da_cprr_2021_v_post_noso_pprr_2020_11_06.xlsx'), decision)
+    matches = matcher.get_index_pairs_within_thresholds(lower_bound=decision)
+    match_dict = dict(matches)
+
+    cprr.loc[:, 'uid'] = cprr.uid.map(lambda x: match_dict.get(x, x))
```

Comment:
Why do you assign uid from post to cprr? What is the purpose?

Reply:
We don't have a pprr for the New Orleans's Sheriff's office. We only have POST and per-new-orleans-so, which was extracted from cprr_new_orleans_so so I thought it was correct to match to POST. 

I now see why it is wrong for NOPD. Is this also wrong for NOSO?

Reply:
If you are going to lift roster data for an agency from POST then it's proper to do this. The point is to have consistent uid across tables. Since we are extracting roster data for NOSO from POST I guess it is okay.